lesson6

dontjke opened the pull request.

dontjke committed:
Animated Vector Drawable

dontjke committed:
Animated Vector Drawable факультатив

dontjke committed:
RecyclerView viewType | добавил макет для элемента

dontjke committed:
RecyclerView viewType | добавил макет для второго элемента

dontjke committed:
добавил модель данных для списка

dontjke committed:
добавил слушатель нажатий на элемент списка

dontjke committed:
добавил макет и активити RecyclerActivity.kt

dontjke committed:
добавил RecyclerActivityAdapter

dontjke committed:
создал произвольный список с данными, адаптер

dontjke committed:
добавил ещё один элемент в menu_bottom_navigation

dontjke committed:
добавил заголовок, доработал адаптер и activity

dontjke committed:
отрефакторил код — вынес метод bind в абстрактный класс

dontjke committed:
добавил divider

dontjke committed:
убрал старые dividers в виде view

dontjke committed:
добавил кнопки удаления/добавления

dontjke committed:
добавил FAB и слушатель, методы для удаления/добавления

dontjke committed:
изменил notifyDataSetChanged()

dontjke committed:
добавил кнопки для перемещения элемента

dontjke committed:
повесил слушатели и добавил методы для перемещения

dontjke committed:
Изменение элемента списка | добавил короткое описание

dontjke committed:
изменил список данных, на список пар, данные и состояние(развёрнут/свёрнут) элемента. Доработал activity и сам адаптер

dontjke committed:
Перетягивание и смахивание элементов | добавил два интерфейса

dontjke committed:
имплементировал интерфейсы в адаптер и холдер

dontjke committed:
добавил класс наследник от ItemTouchHelper.Callback()

dontjke committed:
добавил onChildDraw изменение прозрачности при смещении

dontjke committed:
Drag Handle «Перетаскивание за рукоятку» | добавил в макет

dontjke committed:
добавил интерфейс OnStartDragListener.kt имплементировал

dontjke committed:
добавил серый фон

DeltAagent00 commented on app/src/main/java/com/example/material_design/view/recycler/ItemTouchHelperCallback.kt:
```diff
@@ -0,0 +1,84 @@
+        Int
+    ) {
+        if (actionState != ItemTouchHelper.ACTION_STATE_IDLE) {
+            val itemViewHolder = viewHolder as ItemTouchHelperViewHolder
```
вы делаете небезопасное приведение типов, так есть вероятность поучить крэш. Лучше приводить типы через безопасный оператор as?.

DeltAagent00 commented on app/src/main/java/com/example/material_design/view/recycler/RecyclerActivity.kt:
```diff
@@ -0,0 +1,52 @@
+        ItemTouchHelper(ItemTouchHelperCallback(adapter))
+            .attachToRecyclerView(binding.recyclerView)
+
+
```
Не нужно оставлять пустые строки в коде

dontjke committed:
Payload

dontjke committed:
DiffUtil | добавил к модели данных параметр id,

dontjke committed:
обновил код где создаётся Data
data-класс Change хранит в себе два массива: старый список элементов и новый.

dontjke committed:
добавил DiffUtilCallback
Сравниваею id, затем контент если разный, возвращаю экземпляр класса Change, который и попадает в payload

dontjke committed:
изменил onBindViewHolder. сравниваю старые данные с самыми
свежими и заменяю заголовок элемента, если он изменился

dontjke committed:
Анимации для замены и перемещения

dontjke committed:
кнопка для применения изменений

dontjke committed:
добавил функционал генерирования разных списков
